dataeng/jobs/analytics/WarehouseTransformsDaily.groovy

```diff
@@ -0,0 +1,51 @@
+package analytics
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_parameters
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_log_rotator
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_wrappers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_publishers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_triggers
+
+class WarehouseTransformsDaily{
+    public static def job = { dslFactory, allVars ->
+        dslFactory.job("warehouse-transforms-daily"){
+            logRotator common_log_rotator(allVars)
+            parameters secure_scm_parameters(allVars)
+            parameters {
+                stringParam('WAREHOUSE_TRANSFORMS_URL', allVars.get('WAREHOUSE_TRANSFORMS_URL'), 'URL for the Warehouse Transforms Repo.')
+                stringParam('WAREHOUSE_TRANSFORMS_BRANCH', allVars.get('WAREHOUSE_TRANSFORMS_BRANCH'), 'Branch of Warehouse Transforms to use.')
+                stringParam('MODEL_TAG', allVars.get('MODEL_TAG'), 'Tagged model that will be run.')
+                stringParam('DBT_PROJECT', allVars.get('DBT_PROJECT'), 'dbt project in warehouse-transforms to work on.')
+                stringParam('DBT_PROFILE', allVars.get('DBT_PROFILE'), 'dbt profile from analytics-secure to work on.')
+                stringParam('DBT_TARGET', allVars.get('DBT_TARGET'), 'dbt target from analytics-secure to work on.')
+                stringParam('NOTIFY', allVars.get('NOTIFY','$PAGER_NOTIFY'), 'Space separated list of emails to send notifications to.')
+            }
+            multiscm secure_scm(allVars) << {
+                git {
+                    remote {
+                        url('$WAREHOUSE_TRANSFORMS_URL')
+                        branch('$WAREHOUSE_TRANSFORMS_BRANCH')
+                        credentials('1')
+                    }
+                    extensions {
+                        relativeTargetDirectory('warehouse-transforms')
+                        pruneBranches()
+                        cleanAfterCheckout()
+                    }
+                }
+            }
+            publishers common_publishers(allVars)
+            steps {
+                virtualenv {
+                    pythonName('PYTHON_3.7')
+                    nature("shell")
+                    systemSitePackages(false)
+                    command(
+                        dslFactory.readFileFromWorkspace("dataeng/resources/warehouse-transforms-daily.sh")
```

Comment:
Hmm.  I commented on this, but it didn't show up in my review, so I'll try again:   I think this whole thing should be structured like the SQL scripts stuff, to have the individual jobs be defined in analytics-secure, and have this loop over all the warehouse-transform jobs to create.  Each would provide a name, and a schedule, as well as the various parameters.   Note that nothing here is actually "daily", so that should probably be removed from the shell script, the groovy file, and the job name.   And at present, we have two daily SQL script jobs that run completely independently of each other, and I think we should expect there may be something similar going forward.

Reply:
@brianhw I don't think i follow it completely about what you said here.

Reply:
Sorry, I'll try again.  I think this should be in a file WarehouseTransforms.groovy, that points to dataeng/resources/warehouse-transforms.sh.   It should read from a WAREHOUSE_TRANSFORMS_JOB_EXTRA_VARS.yaml that is structured similarly to SQL_SCRIPTS_JOB_EXTRA_VARS.yaml, since warehouse-transforms jobs are intended to replace sql-scripts jobs.  Each ENVIRONMENT has a job "name" (e.g. "bi-daily", "financial-reporting-daily", etc.), and each of these contains the relevant parameters for that job.  These include MODEL_TAG, DBT_PROJECT, DBT_PROFILE, DBT_TARGET, but also JOB_FREQUENCY.   The code in WarehouseTransforms.groovy would include a wrapper for processing multiple jobs:
{code}
    public static def job = { dslFactory, allVars ->
         allVars.get('ENVIRONMENTS').each { environment, env_config ->
            dslFactory.job("warehouse-transforms-$environment") {
                ...
                triggers common_triggers(allVars, env_config)
{code}

The addition of the trigger and the defining of JOB_FREQUENCY is what gets the job to trigger.  Without that, it will only run manually.

Reply:
@brianhw Thanks for the explanation. 
Since you mentioned another daily job (financial-reporting-daily) running independently of "daily" job, i've added this job here also. 
For now i've taken exact same job schedule for these job that we have in "sql-scripts-{$environment}".

Reply:
By here i mean, added it as environment in analytics-secure.

Reply:
@brianhw It's ready for re-review.